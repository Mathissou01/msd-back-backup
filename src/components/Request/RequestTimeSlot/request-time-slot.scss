@use "src/styles/02-tools/tools.pictos" as p;

.c-RequestTimeSlot {
  display: flex;
  line-height: 32px;

  > span {
    width: 20%;
  }

  &__Slots {
    display: flex;
    width: 80%;

    div {
      width: 50%;

      > span:last-of-type {
        margin-left: 10px;
      }
    }
  }
}
