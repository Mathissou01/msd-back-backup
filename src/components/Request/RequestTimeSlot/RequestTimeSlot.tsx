import { ITimeSlotsDisplay } from "../../../lib/requests";
import "./request-time-slot.scss";

interface IRequestTimeSlotProps {
  timeSlot: ITimeSlotsDisplay;
}

export default function RequestTimeSlot({ timeSlot }: IRequestTimeSlotProps) {
  /* Static Data */
  const labels = {
    separator: ":",
    appointment: "rendez-vous",
  };

  return (
    <div className="c-RequestTimeSlot">
      <span data-testid="time-slot-day">{timeSlot.day}</span>
      <div className="c-RequestTimeSlot__Slots">
        {timeSlot.slots.map((slot, slotIndex) => {
          return (
            <div key={slotIndex}>
              <span data-testid="time-slot-text">
                {slot.slot} {labels.separator}
              </span>
              <span data-testid="time-slot-nb-appointments">
                {slot.nbAppointments} {labels.appointment}
              </span>
            </div>
          );
        })}
      </div>
    </div>
  );
}
