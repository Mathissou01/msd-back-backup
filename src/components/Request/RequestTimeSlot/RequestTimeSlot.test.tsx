import { render, screen } from "@testing-library/react";
import RequestTimeSlot from "./RequestTimeSlot";

const timeSlotsMock = {
  day: "Lundi",
  slots: [
    { slot: "08h00 - 12h00", nbAppointments: 3 },
    { slot: "14h00 - 18h00", nbAppointments: 10 },
  ],
};

describe("RequestTimeSlot", () => {
  it("renders", () => {
    const { container } = render(<RequestTimeSlot timeSlot={timeSlotsMock} />);
    expect(screen.getByTestId("time-slot-day")).toBeInTheDocument();
    expect(screen.getAllByTestId("time-slot-text")).toHaveLength(2);
    expect(screen.getAllByTestId("time-slot-nb-appointments")).toHaveLength(2);
    expect(container).toMatchSnapshot();
  });
});
