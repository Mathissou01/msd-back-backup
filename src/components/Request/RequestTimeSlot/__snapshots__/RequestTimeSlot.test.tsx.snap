// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestTimeSlot renders 1`] = `
<div>
  <div
    class="c-RequestTimeSlot"
  >
    <span
      data-testid="time-slot-day"
    >
      Lundi
    </span>
    <div
      class="c-RequestTimeSlot__Slots"
    >
      <div>
        <span
          data-testid="time-slot-text"
        >
          08h00 - 12h00
           
          :
        </span>
        <span
          data-testid="time-slot-nb-appointments"
        >
          3
           
          rendez-vous
        </span>
      </div>
      <div>
        <span
          data-testid="time-slot-text"
        >
          14h00 - 18h00
           
          :
        </span>
        <span
          data-testid="time-slot-nb-appointments"
        >
          10
           
          rendez-vous
        </span>
      </div>
    </div>
  </div>
</div>
`;
